### PR TITLE
added a function for checking internet connectivity

### PR DESCRIPTION
**Pull request by Garg-Akshat**
added a function for checking internet connectivity in the utils folder, it returns a boolean

### CODE REVIEW

File: pubspec.lock [hunk 1]
```diff
@@ -8,6 +8,20 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "1.0.2"
+  archive:
```
Comment (PrarabdhGarg) [inline pubspec.lock]: Changes to this file should not be committed. Please remove them.

**Review by PrarabdhGarg — CHANGES_REQUESTED** [review]

File: pubspec.lock (hunk 1 above)
Comment (Garg-Akshat) [reply pubspec.lock]: These are automatically generated files imo we should add this to .gitignore.
Everytime packages get is done, these files are generated.

**Review by Garg-Akshat — COMMENTED** [review]
I have removed the changes made in the pubspec.lock file
Please check the other files

**Comment by PrarabdhGarg** [comment]
Good Work.
From next time, make sure to resolve the conversations for which you have made the changes.
Also, at the end of your commits, make sure to add Resolves #<Issue Number> to automatically close the corresponding Issue as well on merging the PR.

**Pull request merged**